packages/relay/src/lib/eth.ts

```diff
@@ -1260,19 +1269,28 @@ export class EthImpl implements Eth {
       `${requestIdPrefix} getTransactionByBlockNumberAndIndex(blockNum=${blockNumOrTag}, index=${transactionIndex})`,
     );
     const blockNum = await this.translateBlockTag(blockNumOrTag, requestIdPrefix);
-    return this.mirrorNodeClient
-      .getContractResults(
+
+    try {
+      const contractResults = await this.mirrorNodeClient.getContractResults(
         { blockNumber: blockNum, transactionIndex: Number(transactionIndex) },
         undefined,
         requestIdPrefix,
-      )
-      .then((contractResults) => formatContractResult(contractResults[0] ?? null))
-      .catch((e: any) => {
-        throw this.common.genericErrorHandler(
-          e,
-          `${requestIdPrefix} Failed to retrieve contract result for blockNum ${blockNum} and index=${transactionIndex}`,
-        );
-      });
+      );
+
+      if (!contractResults[0]) return null;
+
+      const resolvedToAddress = await this.resolveEvmAddress(contractResults[0].to, requestIdPrefix);
```

Comment:
Does `resolveEvmAddress` utilize a cache in it's implementation?
If not we should consider to make future look ups quicker and reduce the number of calls to the Mirror Node